numpy/core/include/numpy/npy_os.h

```diff
@@ -19,12 +19,18 @@
     #define NPY_OS_SOLARIS
 #elif defined(__CYGWIN__)
     #define NPY_OS_CYGWIN
+/* We are on Windows.*/
 #elif defined(_WIN32) || defined(__WIN32__) || defined(WIN32)
```

Comment:
Perhaps it's time to retire `WIN32` and `WIN64`.

As far as I can understand and according to [QM Bites: Understand Windows OS Identification Preprocessor Macros](https://accu.org/journals/overload/24/132/wilson_2223/):
* **`_WIN32`** and **`_WIN64`** are defined **by the compiler**,
* **`WIN32`** and **`WIN64`** are defined **by the user**, to indicate whatever the user chooses them to indicate. They mean 32-bit and 64-bit Windows compilation by convention only.

MSVC has been defining `_WIN32` and `_WIN64` for a long time, provably at least since Visual Studio 2015, and in practice as far as back in the days of 16-bit Windows:
* https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-140
* https://learn.microsoft.com/en-us/windows/win32/winprog64/the-tools

Reply:
I agree. I kept them there because they were already there. @HaoZeke should we remove them?

Reply:
> I agree. I kept them there because they were already there. @HaoZeke should we remove them?

Yup, that would be for the best I believe, please go ahead with that, also, thanks for the review and fix @mahge and @DimitriPapadopoulos

Reply:
> Yup, that would be for the best I believe, please go ahead with that, also, thanks for the review and fix @mahge and @DimitriPapadopoulos

No problem at all. Fixed in 356325c.